File: Assignment/Abstraction/Utils/Symbol.cs
using Assignment.Abstraction.Utils;

namespace Assignment.Abstraction
{
    internal class Symbol
    {
        public Symbol(string name, DataTypes type)
        {
            this.Name = name;
            this.Type = type;
        }

        public string Name { get; }
        public DataTypes Type { get; }
        public Scope Scope { get; set; }
    }
}

File: src/main/java/compiler/abstraction/utils/Symbol.java
package compiler.abstraction.utils;

import compiler.abstraction.DataType;

public class Symbol {
    public final Object value;
    public final String name;
    public final DataType dataType;
    public Scope scope;

    public Symbol(String name, DataType dataType) {
        super();
        this.name = name;
        this.dataType = dataType;
        this.value = null;
    }

    public Symbol(String name, DataType dataType, Object value) {
        super();
        this.name = name;
        this.dataType = dataType;
        this.value = value;
    }
}